lazy_tensor_core/lazy_tensor_core/core/extract_compiled_graph.py

```diff
@@ -39,6 +39,46 @@ def __call__(self, args):
             real_input.append(inp)
         return real_input
 
+class ReturnValueHandler:
+    r"""
+    When _LAZYC._ltc_sync_multi is called on multi tensors, the compiled graph
+    will contain output only for unique tensors - if a tensor appears multiple
+    times in the input to _ltc_sync_multi, only the first occurance matters.
+
+    However from python level, we still expect multi tensors returned with duplciation
+    even if the TS graph dedup the output. e.g. for method:
+
+      def forward(self, a):
+        return a, a
+
+    the TS graph captured by LTC will return a single tensor, but Python method expects 2.
+
+    This class dedup the lazy tensors first to get the index that will be used
+    to duplicate the eager tensors later.
+    """
+    def __init__(self, lazy_out_list):
+        self.index = []
+        self.total_count = len(lazy_out_list)
+
+        tensor_id_to_idx = dict()
+        for dup_idx, lazy_tensor in enumerate(lazy_out_list):
+            uniq_idx = tensor_id_to_idx.get(id(lazy_tensor), None)
+            if uniq_idx is not None:
+                self.index[uniq_idx].append(dup_idx)
+            else:
+                uniq_idx = len(self.index)
```

Comment:
your approach looks correct, but is there already a 'unique id' in the lazy tensor class that we could directly use?

I think the id I am thinking of is the data.unique_id, and there should be one for each lazy tensor even if it is backed by an IR value vs tensor data.

Reply:
@wconstab it probably won't give us much here. we seem to need a mapping index -> unique_index and using `id(x)` to construct it works just as well.

Reply:
yep, I think either use LazyTensor::GetUniqueId or id(tensor) to track the uniqueness should work. They are basically equivalent.